THOMAS-Server/ModuleManager.cs

```diff
@@ -38,23 +38,23 @@ public void Initialize()
         private IThomasElement Instantiate(Type elementType)
         {
             // Ist bereits eine Instanz vorhanden?
-           
-            if (_drivers.Any(e => e.GetType () == elementType)) return _drivers.First(e => e.GetType() == elementType);
-            else if (_sensors.Any(e => e.GetType() == elementType)) return _sensors.First(e => e.GetType() == elementType);
-            else if (_actors.Any(e => e.GetType() == elementType)) return _actors.First(e => e.GetType() == elementType);
-            else if (_controllers.Any(e => e.GetType() == elementType)) return _controllers.First(e => e.GetType() == elementType);
+            if(_drivers.ContainsInstance(elementType)) return _drivers.GetInstance(elementType);
+            if(_sensors.ContainsInstance(elementType)) return _sensors.GetInstance(elementType);
+            if(_actors.ContainsInstance(elementType)) return _actors.GetInstance(elementType);
+            if(_controllers.ContainsInstance(elementType)) return _controllers.GetInstance(elementType);
 
             // Parameter, die der neuen Instanz übergeben werden
             List<object> instanceParameters = new List<object>();
 
+            // Erforderliche Abhängigkeiten. Sollte eine nicht verfügbar sein, ist dieses Element auch nicht verfügbar.
             RequirementAttribute requirementAttribute = (RequirementAttribute)elementType.GetCustomAttributes(typeof(RequirementAttribute), true).FirstOrDefault();
-            if (requirementAttribute != null)
+            if(requirementAttribute != null)
```

Comment:
hm, bei sowas fragt man am besten immer den CodeStyle der .Net-sources :D https://referencesource.microsoft.com/#System/parent/parent/parent/InternalApis/NDP_Common/inc/PinnableBufferCache.cs,550
Der sagt mit. In meinem Fall war das nicht der resharper-cleanupper, ich nutze auch den VS-Cleanup, hab den da aber umkonfiguriert.. ich mag den cleanupper von resharper nicht, der macht manchmal mist..
ich fixe das dann auch eben im ganzen projekt.